File: padlock-presenter/src/main/java/com/pyamsoft/padlock/presenter/purge/PurgeModule.java
/*
 * Copyright 2016 Peter Kenji Yamanaka
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.pyamsoft.padlock.presenter.purge;

import dagger.Module;
import dagger.Provides;
import javax.inject.Named;
import rx.Scheduler;

@Module public class PurgeModule {

  @Provides PurgePresenter providePinEntryPresenter(final PurgeInteractor interactor,
      @Named("obs") Scheduler obsScheduler, @Named("sub") Scheduler subScheduler) {
    return new PurgePresenterImpl(interactor, obsScheduler, subScheduler);
  }

  @Provides PurgeInteractor providePinEntryInteractor(final PurgeInteractorImpl interactor) {
    return interactor;
  }
}

File: padlock-purge/src/main/java/com/pyamsoft/padlock/purge/PurgeModule.kt
/*
 *     Copyright (C) 2017 Peter Kenji Yamanaka
 *
 *     This program is free software; you can redistribute it and/or modify
 *     it under the terms of the GNU General Public License as published by
 *     the Free Software Foundation; either version 2 of the License, or
 *     (at your option) any later version.
 *
 *     This program is distributed in the hope that it will be useful,
 *     but WITHOUT ANY WARRANTY; without even the implied warranty of
 *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *     GNU General Public License for more details.
 *
 *     You should have received a copy of the GNU General Public License along
 *     with this program; if not, write to the Free Software Foundation, Inc.,
 *     51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 */

package com.pyamsoft.padlock.purge

import com.pyamsoft.padlock.model.PurgeAllEvent
import com.pyamsoft.padlock.model.PurgeEvent
import com.pyamsoft.padlock.api.PurgeInteractor
import com.pyamsoft.pydroid.bus.EventBus
import com.pyamsoft.pydroid.data.Cache
import dagger.Binds
import dagger.Module
import javax.inject.Named

@Module
abstract class PurgeModule {

    @Binds
    internal abstract fun providePurgeBus(bus: PurgeBus): EventBus<PurgeEvent>

    @Binds
    internal abstract fun providePurgeAllBus(bus: PurgeAllBus): EventBus<PurgeAllEvent>

    @Binds
    internal abstract fun providePurgeInteractorCache(impl: PurgeInteractorCache): PurgeInteractor

    @Binds
    @Named("interactor_purge")
    internal abstract fun providePurgeInteractor(impl: PurgeInteractorImpl): PurgeInteractor

    @Binds
    @Named("cache_purge")
    internal abstract fun provideCache(impl: PurgeInteractorCache): Cache
}
